federated_aws_cli/login.py

```diff
@@ -62,11 +59,46 @@ def generate_challenge(code_verifier):
         hashlib.sha256(code_verifier.encode()).digest())
 
 
+def open_web_console(credentials):
+    logger.debug("Attempting to open AWS console.")
+
+    creds = {
+        "sessionId": credentials["AccessKeyId"],
+        "sessionKey": credentials["SecretAccessKey"],
+        "sessionToken": credentials["SessionToken"],
+    }
+
+    params = urlencode({
+        "Action": "getSigninToken",
+        "Session": json.dumps({
+            "sessionId": credentials["AccessKeyId"],
+            "sessionKey": credentials["SecretAccessKey"],
+            "sessionToken": credentials["SessionToken"]
+        }),
+    })
+
+    logger.debug("Web Console params: {}".format(params))
+
+    url = "https://signin.aws.amazon.com/federation?Action=getSigninToken"
+    url += "&Session={}".format(quote_plus(json.dumps(creds)))
+
+    token = requests.get(url).json()
+
+    url = "https://signin.aws.amazon.com/federation?Action=login"
+    url += "&Destination=" + quote_plus("https://console.aws.amazon.com/")
+    url += "&SigninToken=" + token["SigninToken"]
```

Comment:
Have you thought about using the [`urlparse`](https://docs.python.org/2/library/urlparse.html) (or [`urllib.parse`](https://docs.python.org/3/library/urllib.parse.html)) builtin modules to construct these URLs instead of string concatenation?

Reply:
If you're able to make it work with `urlencode` that would be amazing, but I never had any luck.